corehq/apps/case_search/tests/test_filter_dsl.py

```diff
@@ -394,3 +398,26 @@ def test_nested_parent_lookups(self):
         built_filter = build_filter_from_ast(self.domain, parsed)
         self.assertEqual(expected_filter, built_filter)
         self.assertEqual([self.child_case_id], CaseSearchES().filter(built_filter).values_list('_id', flat=True))
+
+
+class TestGetProperties(SimpleTestCase):
+    pass
+
+
+@generate_cases([
+    # equality
+    ("property = 'value'", ['property']),
+    # comparison
+    ("property > 100", ['property']),
+    # complex expression
+    ("first_property > 100 or second_property = 'foo' and third_property = 'bar'",
+     ['first_property', 'second_property', 'third_property']),
+    # malformed expression
+    ("foo = 'bar' or baz = buzz or ham = 'spam' and eggs", ['foo', 'baz', 'ham']),
+    # related case lookup
+    ("parent/parent/foo = 'bar' and parent/baz = 'buzz'", ['parent/parent/foo', 'parent/baz']),
+    # duplicate properties
+    ("property = 'value' or property = 'other_value'", ['property']),
+], TestGetProperties)
+def test_get_properties_from_ast(self, expression, expected_values):
```

Comment:
I am a little confused how this test works. Is it part of a class that will be run? Where does `expected_values` come from

Reply:
this uses `@generate_cases`:
https://github.com/dimagi/commcare-hq/blob/865010a80ee56d210b7da4e3e5ae3e721ebf2be2/corehq/util/test_utils.py#L304-L325

It runs on the `TestGetProperties` class, above.